File: factors.c
#include <stdio.h>
#include <stdlib.h>

int getFactors(int givenNumber)
{
	int factor1 = 0;
	int factor2 = 0;

	for (int number = 2; number < givenNumber; number++)
	{
		if (givenNumber % number == 0)
		{
			factor1 = givenNumber / number;
			factor2 = number;
			break;
		}
	}

	printf("%d=%d*%d\n", givenNumber, factor1, factor2);
	return (0);
}

int factorizeFromFile(const char *filename)
{
	FILE *file = fopen(filename, "r");

	if (file == NULL)
	{
		printf("File '%s' not found.\n", filename);
		return (1);
	}

	char line[100];

	while (fgets(line, sizeof(line), file))
	{
		int number = atoi(line);

		getFactors(number);
	}

	fclose(file);
	return (0);
}

int main(int argc, char *argv[])
{
	if (argc != 2)
	{
		printf("Usage: %s <filename>\n", argv[0]);
		return (1);
	}
	else
	{
		const char *filename = argv[1];

		return (factorizeFromFile(filename));
	}
}

File: factors.py
#!/usr/bin/python3
import sys
import time

def getFactors(givenNumber):
    factor1 = 0
    factor2 = 0
    for number in range(2, givenNumber):
        if givenNumber % number == 0:
            factor1 = givenNumber // number
            factor2 = number
            break

    return f'{givenNumber}={factor1}*{factor2}'

def factorize_from_file(filename):
    try:
        with open(filename, 'r') as file:
            start_time = time.time()
            for line in file:
                number = int(line.strip())
                result = getFactors(number)
                print(result)
            end_time = time.time()
            elapsed_time = end_time - start_time
            print(f"Script execution time: {elapsed_time:.4f} seconds")
    except FileNotFoundError:
        print(f"File '{filename}' not found.")
    except ValueError:
        print(f"Invalid data in '{filename}'. Please make sure each line contains a valid number.")

if __name__ == '__main__':
    if len(sys.argv) != 2:
        print("Usage: ./factor.py <filename>")
    else:
        filename = sys.argv[1]
        factorize_from_file(filename)
